tests/jvm/src/test/scala/cats/tests/FutureTests.scala

```diff
@@ -14,27 +15,29 @@ import org.scalacheck.Arbitrary.arbitrary
 class FutureTests extends CatsSuite {
   val timeout = 3.seconds
 
-  implicit val eqkf: EqK[Future] =
-    new EqK[Future] {
-      def synthesize[A: Eq]: Eq[Future[A]] = futureEq(timeout)
-    }
-
   def futureXor[A](f: Future[A]): Future[Xor[Throwable, A]] =
     f.map(Xor.right[Throwable, A]).recover { case t => Xor.left(t) }
 
-  implicit val eqv: Eq[Future[Int]] =
-    new Eq[Future[Int]] {
-      implicit val throwableEq: Eq[Throwable] = Eq.fromUniversalEquals
-
-      def eqv(x: Future[Int], y: Future[Int]): Boolean =
-        futureEq[Xor[Throwable, Int]](timeout).eqv(futureXor(x), futureXor(y))
+  implicit val eqkf: EqK[Future] =
+    new EqK[Future] {
+      def synthesize[A: Eq]: Eq[Future[A]] =
+        new Eq[Future[A]] {
+          def eqv(fx: Future[A], fy: Future[A]): Boolean = {
+            val fz = futureXor(fx) zip futureXor(fy)
+            Await.result(fz.map { case (tx, ty) => tx === ty }, timeout)
+          }
+        }
     }
 
+  implicit val throwableEq: Eq[Throwable] =
+    Eq.fromUniversalEquals
+
   implicit val comonad: Comonad[Future] = futureComonad(timeout)
 
   // Need non-fatal Throwables for Future recoverWith/handleError
   implicit val nonFatalArbitrary: Arbitrary[Throwable] =
-    Arbitrary(arbitrary[Exception].map(e => e.asInstanceOf[Throwable]))
+    //Arbitrary(arbitrary[Exception].map(_.asInstanceOf[Throwable]))
```

Comment:
Leftover comment? And reason for using a const generator instead of this one? (though in practice it may not matter)